src/vm/coreassemblyspec.cpp

```diff
@@ -183,23 +183,35 @@ VOID  AssemblySpec::Bind(AppDomain      *pAppDomain,
                           &pPrivAsm);
     }
 
+    ICLRPrivAssembly *pBoundAssembly = NULL;
+
     if(SUCCEEDED(hr))
     {
         _ASSERTE(pPrivAsm != nullptr);
-        result = BINDER_SPACE::GetAssemblyFromPrivAssemblyFast(pPrivAsm.Extract());
+        pBoundAssembly = pPrivAsm.Extract();
+        result = BINDER_SPACE::GetAssemblyFromPrivAssemblyFast(pBoundAssembly);
         _ASSERTE(result != nullptr);
     }
     
     pResult->SetHRBindResult(hr);
     if (SUCCEEDED(hr))
     {
-        BOOL fIsInGAC = pAppDomain->IsImageFromTrustedPath(result->GetNativeOrILPEImage());
+        BOOL fIsInGAC = FALSE;
         BOOL fIsOnTpaList = FALSE;
-        const SString &sImagePath = result->GetNativeOrILPEImage()->GetPath();
-        if (pTPABinder->IsInTpaList(sImagePath))
+
+        _ASSERTE(pBoundAssembly != NULL);
+        
+        // Only initialize TPA/GAC status if we bound using DefaultContext
+        if (AreSameBinderInstance(pTPABinder, reinterpret_cast<ICLRPrivBinder *>(pBoundAssembly)))
         {
-            fIsOnTpaList = TRUE;
+            pAppDomain->IsImageFromTrustedPath(result->GetNativeOrILPEImage());
```

Comment:
Delete the call to `IsImageFromTrustedPath` ? It does not seem to be doing anything useful.